addons/mail/static/tests/new/activity/activity_mark_as_done_popover_tests.js

```diff
@@ -0,0 +1,53 @@
+/** @odoo-module **/
+
+import { ActivityMarkAsDone } from "@mail/new/activity/activity_markasdone_popover";
+import { activityData } from "./activityData";
+import { click, getFixture, mount, patchWithCleanup } from "@web/../tests/helpers/utils";
+import { makeTestEnv, TestServer } from "../helpers/helpers";
+
+let target;
+
+QUnit.module("mail", (hooks) => {
+    hooks.beforeEach(async () => {
+        target = getFixture();
+    });
+
+    QUnit.module("activity_mark_as_done_popover");
+
+    QUnit.test("Activity mark as done popover", async (assert) => {
+        const server = new TestServer();
+        const env = makeTestEnv((route, params) => {
+            if (
+                [
+                    "/web/dataset/call_kw/mail.activity/action_feedback",
+                    "/mail/thread/messages",
+                ].includes(route)
+            ) {
+                assert.step(route);
+            }
+            return server.rpc(route, params);
+        });
+        patchWithCleanup(env.services.action, {
+            doAction(action, options) {
+                assert.step("open_activity_form_view");
+            },
+        });
+        await mount(ActivityMarkAsDone, target, {
+            env,
+            props: { activity: activityData },
+        });
+        await click(document.querySelector(".o-mail-activity-mark-as-done-button-done"));
+        assert.verifySteps([
+            "/web/dataset/call_kw/mail.activity/action_feedback",
+            "/mail/thread/messages",
+        ]);
+        await click(
+            document.querySelector(".o-mail-activity-mark-as-done-button-done-and-schedule")
+        );
+        assert.verifySteps([
+            "/web/dataset/call_kw/mail.activity/action_feedback",
+            "/mail/thread/messages",
+            "open_activity_form_view",
+        ]);
+    });
```

Comment:
Prefer making a chatter test.
This would let you properly add activity data in `TestServer`, and let chatter code fetch it.
This would also make it clearer where `ActivityMarkAsDone` popover is used, make it clearer some assertions (`/mail/thread/messages` leaks chatter concerns), and also have better test coverage of the feature.

Doing so will likely give motivation for making more than a single test. For example, it's unclear why `/mail/thread/messages` during mark as done is relevant in this test. If you make another test that asserts activity is turned into a "mark as done" message, then it's very clear why we actively fetch messages.